File: 프로그래머스/lv1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.js
function solution(s) {
    var answer = "";
    let catchNum = [
        [`one`, `1`],
        [`two`, `2`],
        [`three`, `3`],
        [`four`, `4`],
        [`five`, `5`],
        [`six`, `6`],
        [`seven`, `7`],
        [`eight`, `8`],
        [`nine`, `9`],
        [`zero`, `0`]
    ];

    for(let i = 0 ; i < catchNum.length ; i++){
        s = s.replaceAll(`${catchNum[i][0]}`,`${catchNum[i][1]}`);
    }
    answer = parseInt(s);
    return answer;
}

File: 프로그래머스/lv1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.java
class Solution {
    public int solution(String s) {
        int answer = 0;
        String changeStr = s;

        String[] str = {"zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"};
        for(int i = 0; i < 10 ; i++){
           changeStr = changeStr.replace(str[i], Integer.toString(i));
        }

        answer = Integer.parseInt(changeStr);

        return answer;
    }
}